domain/src/test/java/org/oppia/android/domain/oppialogger/loguploader/LogUploadWorkerTest.kt

```diff
@@ -402,6 +413,95 @@ class LogUploadWorkerTest {
     assertThat(currentStatus).isEqualTo(NO_CONNECTIVITY)
   }
 
+  @Test
+  fun testWorker_logFirestoreEvent_withNetwork_enqueueRequest_verifySuccess() {
+    setUpTestApplicationComponent()
+    networkConnectionUtil.setCurrentConnectionStatus(NONE)
+    dataController.logEvent(
+      createOptionalSurveyResponseContext(),
+      profileId = null,
+      1556094120000
+    )
+    networkConnectionUtil.setCurrentConnectionStatus(LOCAL)
+    testCoroutineDispatchers.runCurrent()
+
+    val workManager = WorkManager.getInstance(ApplicationProvider.getApplicationContext())
+
+    val inputData = Data.Builder().putString(
+      LogUploadWorker.WORKER_CASE_KEY,
+      LogUploadWorker.FIRESTORE_WORKER
+    ).build()
+
+    val request: OneTimeWorkRequest = OneTimeWorkRequestBuilder<LogUploadWorker>()
+      .setInputData(inputData)
+      .build()
+
+    workManager.enqueue(request)
+    testCoroutineDispatchers.runCurrent()
+    val workInfo = workManager.getWorkInfoById(request.id)
+
+    assertThat(workInfo.get().state).isEqualTo(WorkInfo.State.SUCCEEDED)
+    assertThat(fakeFirestoreEventLogger.getMostRecentEvent()).isEqualTo(
+      optionalSurveyResponseEventLog
+    )
+  }
+
+  @Test
+  fun testWorker_logFirestoreEvent_withoutNetwork_enqueueRequest_writeFails_verifyFailure() {
+    setUpTestApplicationComponent()
+    networkConnectionUtil.setCurrentConnectionStatus(NONE)
+    dataController.logEvent(
+      createOptionalSurveyResponseContext(),
+      profileId = null,
+      1556094120000
+    )
+    testCoroutineDispatchers.runCurrent()
+
+    val workManager = WorkManager.getInstance(ApplicationProvider.getApplicationContext())
+
+    val inputData = Data.Builder().putString(
+      LogUploadWorker.WORKER_CASE_KEY,
+      LogUploadWorker.FIRESTORE_WORKER
+    ).build()
+
+    val request: OneTimeWorkRequest = OneTimeWorkRequestBuilder<LogUploadWorker>()
+      .setInputData(inputData)
+      .build()
+
+    setUpFirestoreEventLoggerToFail()
+    workManager.enqueue(request)
+    testCoroutineDispatchers.runCurrent()
+    val workInfo = workManager.getWorkInfoById(request.id)
+
+    // assertThat(workInfo.get().state).isEqualTo(WorkInfo.State.FAILED)
```

Comment:
Please remove any commented out code included in the PR.

Reply:
Uncommented this assertion. I previously missed this.